src/EditorFeatures/DiagnosticsTestUtilities/CodeActions/CSharpCodeRefactoringVerifier`1+Test.cs

```diff
@@ -69,6 +69,14 @@ public Test()
             /// <inheritdoc cref="SharedVerifierState.Options"/>
             internal OptionsCollection Options => _sharedState.Options;
 
+#if !CODE_STYLE
+            internal CodeActionOptionsProvider CodeActionOptions
+            {
+                get => _sharedState.CodeActionOptions;
+                set => _sharedState.CodeActionOptions = value;
+            }
+#endif
```

Comment:
Instead of this, try having the following override:

```csharp
			protected override Workspace CreateWorkspaceImpl()
			{
				var workspace = base.CreateWorkspaceImpl();
				workspace.Options = workspace.Options.WithChangedOption(new OptionKey(CodeActionOptionsStorage.EnableConvertToRecord), true);
				return workspace;
			}
```

Reply:
I noticed the method already has an override. You'll need to include the `Options` change there both in the in-process and the out-of-process code paths

Reply:
Unfortunately we need another hook -- since we're moving off of workspace.Options being used directly in features, the newer pattern reads from the CodeActionsOptionsProvider and needs a redirect there...

Reply:
Do we need FixAll support in testing library to be able to do that?

This is how it work for codefixes:

https://github.com/dotnet/roslyn/blob/2c04f45620a254a18b9c7dd61e73f76e5974c791/src/EditorFeatures/DiagnosticsTestUtilities/CodeActions/CSharpCodeFixVerifier%602%2BTest.cs#L114-L134

We pass `_sharedState.CodeActionOptions` to `FixAllState`

Reply:
That'd be the pattern that we're introducing here, so looks like that's already correct.